mlrun/runtimes/nuclio/application/application.py

```diff
@@ -429,11 +426,33 @@ def create_api_gateway(
         self,
         name: str = None,
```

Comment:
`name` is not enriched, can it still be `None`?

Reply:
it will fail in `APIGatewayMetadata` constructor

Reply:
IMO better to have it as a required arg here instead of failing later